core/src/main/java/com/graphhopper/routing/util/countryrules/CountryRule.java

```diff
@@ -19,15 +19,15 @@
 package com.graphhopper.routing.util.countryrules;
 
 import com.graphhopper.reader.ReaderWay;
-import com.graphhopper.routing.ev.RoadAccess;
+import com.graphhopper.routing.ev.CarRoadAccess;
 import com.graphhopper.routing.ev.Toll;
 import com.graphhopper.routing.util.TransportationMode;
 
 /**
  * GraphHopper uses country rules to adjust the routing behavior based on the country an edge is located in
  */
 public interface CountryRule {
-    default RoadAccess getAccess(ReaderWay readerWay, TransportationMode transportationMode, RoadAccess currentRoadAccess) {
+    default CarRoadAccess getAccess(ReaderWay readerWay, TransportationMode transportationMode, CarRoadAccess currentRoadAccess) {
```

Comment:
```java
getCarAccess(ReaderWay, CarRoadAccess)
```

Otherwise we would need to come up with a generic approach for the various RoadAccess enums in order to make use of the passed TransportationMode.

Reply:
btw why do we want to split the RoadAccess enum? I understand that some values only make sense for certain transportation modes but keeping them in a single enum keeps interfaces like this one a lot cleaner.